### PR TITLE
Update Pi-hole from browser

### PR DESCRIPTION
**Pull request by AlexLamond**
**By submitting this pull request, I confirm the following:** `{please fill any appropriate checkboxes, e.g: [X]}`

`{Please ensure that your pull request is for the 'devel' branch!}`

- [X] I have read and understood the [contributors guide](https://github.com/pi-hole/AdminLTE/blob/master/CONTRIBUTING.md), as well as this entire template.
- [X] I have made only one major change in my proposed changes.
- [X] I have commented my proposed changes within the code.
- [X] I have tested my proposed changes.
- [X] I am willing to help maintain this change if there are issues with it later.
- [X] I give this submission freely and claim no ownership.
- [X] It is compatible with the [EUPL 1.2 license](https://opensource.org/licenses/EUPL-1.1)
- [X] I have squashed any insignificant commits. ([`git rebase`](http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html))
- [X] I have Signed Off all commits. ([`git commit --signoff`](https://git-scm.com/docs/git-commit#Documentation/git-commit.txt---signoff))

---

**What does this PR aim to accomplish?:**

This PR aims to defeat one of the hurdles in keeping Pi-hole up-to-date by adding an update pane in the Pi-hole web interface

image

**How does this PR accomplish the above?:**

This is actually accomplished quite easily, it uses the existing code from the gravity update page and from there modifications have been made to ensure it runs pihole -up

**What documentation changes (if any) are needed to support this PR?:**

May need documentation on updating Pi-hole from inside the web browser. Not sure if it would also make sense to add it to the footer when an update is available to point to the update page

**How does this PR accomplish the above?:**

This is actually accomplished quite easily, it uses the existing code from the gravity update page and from there modifications have been made to ensure it runs pihole -up

**What documentation changes (if any) are needed to support this PR?:**

May need documentation on updating Pi-hole from inside the web browser. Not sure if it would also make sense to add it to the footer when an update is available to point to the update page

### CODE REVIEW

**Comment by dschaper** [comment]
What happens when the update changes `lighttpd` and requires a restart of the daemon?

**Comment by AlexLamond** [comment]
Not sure I'm afraid. Is there a way to trigger that happening? It's slightly had to test for events that I can't reproduce easily or don't know how to reproduce

**Comment by PromoFaux** [comment]
You might get away with doing a hard reset of the local repo for the core repository to force it to be on the previous released tag

```
cd /etc/.pihole
sudo git reset --hard v5.5
```

`lighttpd` is pretty much always restarted when the core repo updates - so this is probably where you're going to come up against a hurdle. I think we may have been down this road before... but no harm in a fresh pair of eyes on the challenge!

**Comment by yubiuser** [comment]
How do you get users to read the release notes if you make updating so easy? They **will** break their system at one point (major update with breaking changes). DNS is a crucial network component and any change to it (so is updating) should be made well prepared and knowing what changes will occur.

**Comment by DL6ER** [comment]
We tried this before and restarting the webserver was the only issue. The code for this may still be around somewhere. This will become a lot easier when we have (a) FTL be the web server and (b) have a native systemd unit that can auto-restart FTL if needed.

**Comment by AlexLamond** [comment]
I'm currently working on a system that uses a JSON function to read from a file that's based on the proc command running a bash script that will not disappear due to the PHP server rebooting

File: update.php [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+<div class="page-header">
+    <h1>Update PiHole</h1>
+</div>
+
+<!-- Alerts -->
+<div id="alInfo" class="alert alert-info alert-dismissible fade in" role="alert" hidden>
+    <button type="button" class="close" data-hide="alert" aria-label="Close"><span aria-hidden="true">&times;</span></button>
+    Updating... this may take a while. <strong>Please do not navigate away from or close this page.</strong>
+</div>
+<div id="alSuccess" class="alert alert-success alert-dismissible fade in" role="alert" hidden>
```
Comment (GrantBirki) [inline update.php:20]: I see there is an `updating` and `success` section that gets displayed as a valid alert. Do you have any logic that sends back an error message if something fails?

Comment (AlexLamond) [reply update.php:20]: This was going to be fixed in a future pull request, initially this is a POC

**Comment by rdwebdesign** [comment]
One more point that needs attention:

How will this work in a docker installation?
Will it be automatically disabled?

**Pull request closed without merging**